test/bench/portfft/launch_bench.hpp

```diff
@@ -56,9 +57,11 @@ void bench_dft_average_host_time_impl(benchmark::State& state, sycl::queue q, po
   double ops = cooley_tukey_ops_estimate(N, N_transforms);
   std::size_t bytes_transferred = global_mem_transactions<complex_type, complex_type>(N_transforms, N, N);
 
-  forward_t* in_dev = sycl::malloc_device<forward_t>(num_elements, q);
-  complex_type* out_dev =
-      desc.placement == portfft::placement::IN_PLACE ? nullptr : sycl::malloc_device<complex_type>(num_elements, q);
+  auto in_dev = make_shared<forward_t>(num_elements, q);
+  std::shared_ptr<complex_type> out_dev;
+  if (desc.placement == portfft::placement::OUT_OF_PLACE) {
+    out_dev = make_shared<complex_type>(num_elements, q);
+  }
```

Comment:
There is anther comment about an identical pattern to this.

Reply:
Done in https://github.com/codeplaysoftware/portFFT/pull/124/commits/938491de23c83c1850b4a91aef39c97acf357c42